cpp/src/arrow/type.h

```diff
@@ -475,7 +496,7 @@ class ARROW_EXPORT NullType : public DataType {
   std::string ToString() const override;
 
   DataTypeLayout layout() const override {
-    return {{DataTypeLayout::kAlwaysNullBuffer}, false};
+    return DataTypeLayout({DataTypeLayout::AlwaysNull()});
```

Comment:
Is it a failure to pass `{}` vs `{nullptr}` in Null arrays buffers?

Reply:
I think so, at least as the current Arrow codebase.

Reply:
I think that is the case